ingest/writer.go

```diff
@@ -301,10 +346,40 @@ func (f *FileWriter) Write(data string) error {
 		return err
 	}
 
+	var compressedData []byte
+	switch f.CompressionMethod {
+	case "gzip":
+		var buff bytes.Buffer
```

Comment:
Can we avoid creating a new buffer and instead write directly to the file? You should be able to say `gzip.NewWriter(f.fd)` for a similar effect. Here's an example, and some discussion as well https://gist.github.com/mchirico/6147687